flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlChunkSplitter.java

```diff
@@ -148,7 +148,12 @@ private void analyzeTable(TableId tableId) {
                     ChunkUtils.getChunkKeyColumn(
                             currentSplittingTable, sourceConfig.getChunkKeyColumn());
             splitType = ChunkUtils.getChunkKeyColumnType(splitColumn);
-            minMaxOfSplitColumn = queryMinMax(jdbcConnection, tableId, splitColumn.name());
+            // split column would be null if table has no primary key.
+            if (splitColumn != null) {
+                minMaxOfSplitColumn = queryMinMax(jdbcConnection, tableId, splitColumn.name());
+            } else {
+                minMaxOfSplitColumn = new Object[2];
+            }
```

Comment:
The chunk splitter should return a total split immediately when the table does not have the primary key. This part should not be changed.